pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -688,6 +688,16 @@ func getDataSecAtRestInfo(ctx *zedagentContext) *info.DataSecAtRest {
 	return ReportDataSecAtRestInfo
 }
 
+func addConfigItemsToDeviceInfo(deviceInfoPtr *info.ZInfoDevice,
+	globalConfigPtr *types.GlobalConfig) {
+	cfgItemsPtr := new(info.ZInfoConfigItems)
+	// XXX - Add only Errors for now. ConfigItems need a bit of restructuring
+	//  to avoid requiring repeat code all over the place.
+	cfgItemsPtr.Errors = make([]string, len(globalConfigPtr.Errors))
+	_ = copy(cfgItemsPtr.Errors, globalConfigPtr.Errors)
+	deviceInfoPtr.ConfigItems = cfgItemsPtr
```

Comment:
This is what confused me; didn't realize it was setting a field in the protobuf message.

I'd suggest the function return a configItemsPtr, and the caller set the field in the message.
That is more logical to me.